wled00/json.cpp

```diff
@@ -112,8 +112,8 @@ bool deserializeSegment(JsonObject elem, byte it, byte presetId)
   if (stop > start && of > len -1) of = len -1;
 
   // update segment (delete if necessary)
-  // we must not change segment dimensions during drawing of effects as that may produce undesired behaviour (crash)
-  seg.setUp(start, stop, grp, spc, of, startY, stopY, !strip.isServicing());
+  // we must not change segment dimensions during drawing of effects in that segment as concurrent access may cause a crash
+  seg.setUp(start, stop, grp, spc, of, startY, stopY, id);
```

Comment:
you can check here which segment is active (being drawn to) as `id == strip.getCurrSegmentId()` to force ID detection outside of Segment class

Reply:
I do not want to have this logic in json.cpp as queuing should be considered internal WS2812FX and Segment functionality, it has nothing to do with JSON per se. This check would also need to be duplicated in `set.cpp` for the legacy HTTP API.

Reply:
I'm not sure which is uglier: 
`seg.setUp(start, stop, grp, spc, of, startY, stopY, id == strip.getCurrSegmentId());`
or
`strip.setSegment(id, start, stop, grp, spc, of, startY, stopY);` (with added queueing in setSegment)

 😉 

And BTW `set.cpp` only ever deals with one segment at a time, but you are correct and I missed that. Still, the original code forced queued segment change which was a desired thing.